File: Project-1-CrimeIntel-AI-Chatbot/frontend/src/services/reportService.js
import api from "./api";

export const createPDFReport = async (
  reportTitle,
  caseMasterId = null,
  reportType = "case_report"
) => {
  const response = await api.post("/api/reports/pdf", {
    case_master_id: caseMasterId,
    report_title: reportTitle,
    report_type: reportType,
  });

  return response.data;
};

export const getReports = async () => {
  const response = await api.get("/api/reports/");
  return response.data;
};

File: Project-1-CrimeIntel-AI-Chatbot/backend/app/services/report_service.py
from sqlalchemy.orm import Session

from app.models.ai_models import PDFReport
from app.schemas.report_schema import PDFReportRequest


def create_pdf_report_placeholder(
    db: Session,
    report_request: PDFReportRequest
):
    file_path = f"reports/{report_request.report_title.replace(' ', '_').lower()}.pdf"

    report = PDFReport(
        case_master_id=report_request.case_master_id,
        report_title=report_request.report_title,
        report_type=report_request.report_type,
        file_path=file_path
    )

    db.add(report)
    db.commit()
    db.refresh(report)

    return report


def get_all_reports(db: Session):
    return db.query(PDFReport).order_by(PDFReport.created_at.desc()).all()